igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy

```diff
@@ -134,123 +133,87 @@ class JenkinsBuildMonitor implements PollingMonitor {
         }
     }
 
-    /*
-     * retrieves a list of new builds that are different than the ones in cache and keeps track of the builds it has
+    /**
+     * Gets a list of jobs for this master & processes builds between last poll stamp and a sliding upper bound stamp
+     * Advances the cursor to the upper bound when all builds are completed
+     * Post an event per completed build
+     * @param master: a jenkins master
      */
 
-    List<Map> changedBuilds(String master) {
-
-        log.info('Checking for new builds for ' + master)
-        List<Map> results = []
+    void changedBuilds(String master) {
+        log.info("Checking for new builds for ${master}")
+        def startTime = System.currentTimeMillis()
+        lastPoll = startTime
 
-        lastPoll = System.currentTimeMillis()
-        long thisPoll = lastPoll
         try {
-            List<String> cachedBuilds = cache.getJobNames(master)
-
-            def startTime = System.currentTimeMillis()
-            List<Project> builds = buildMasters.map[master].projects?.list
+            JenkinsService jenkinsService = buildMasters.map[master] as JenkinsService
+            List<Project> jobs = jenkinsService.getProjects()?.getList() ?:[]
+            log.info("Jobs on master: ${master} ${jobs*.name}")
+
+            for (Project job : jobs) {
+                if (!job.lastBuild) {
+                    log.info("no builds on ${master}:${job.name} skipping...")
+                    continue
+                }
+
+                Long cursor = cache.getLastPollCycleTimestamp(master, job.name)
+                Long lastBuildStamp = job.lastBuild.timestamp as Long
+                Date upperBound = new Date(lastBuildStamp)
+
+                log.info("last cursor was ${cursor}, last build on this job was at ${upperBound}")
```

Comment:
The logs for this will be really noisy, can we clean it up so it logs the job name & build number only when there is a hit?